### DISCUSSION TITLE
After configuring Keycloak authentication in Quarkus, access is intercepted and returns a 401 or 403, unable to redirect to the Keycloak login page.

### DISCUSSION THREAD

**Discussion by ystxdong** [quarkusio/quarkus#39045] [Community]
1.After Keycloak authentication, intercepted requests cannot be redirected to the Keycloak login page.
2.In Quarkus configuration, is it possible not to use quarkus.oidc.credentials.secret? If removed, it will result in an error.

ps: I am a beginner and look forward to expert advice.

**Comment by sberyozkin** [comment]
@ystxdong Can you try https://quarkus.io/guides/security-oidc-code-flow-authentication-tutorial ? 

> is it possible not to use quarkus.oidc.credentials.secret

It is possible for OIDC `service` endpoints but since, I believe, you'd like to use the authorization code flow involving the redirect to Keycloak, the secret is usually needed (or some other supported form of the OIDC client authentication). Without a password it is a public client

**Reply by ystxdong** [reply]
Thank you very much, I was successfully redirected to the login page following the provided URL. How do I configure a public client?

**Reply by sberyozkin** [reply]
@ystxdong Please check Keycloak docs, I guess you just don't set a secret.
